AssertionError in rebindSuper

## Minimized code

```Scala
// AE-ed9d1b1f2a3006febec44cbf09eca6d2226b5b0f
trait Parent {
  def name: String
}
trait Son extends Parent {
  abstract override def name = ""
  def parentName = super.name
}
class GrandSon extends Son
```

If I remove `abstract`, I get:
```scala
-- Error: test.scala:6:25 ------------------------------------------------------
6 |  def parentName = super.name
  |                   ^^^^^^^^^^
  |method name in trait Parent is accessed from super. It may not be abstract unless it is overridden by a member declared `abstract' and `override'
```
I think this requires further elaboration 😅 🤔 

## Output (click arrow to expand)
<details>

```scala
exception occurred while compiling test.scala
java.lang.AssertionError: assertion failed while compiling test.scala
Exception in thread "main" java.lang.AssertionError: assertion failed
	at dotty.DottyPredef$.assertFail(DottyPredef.scala:16)
	at dotty.tools.dotc.transform.ResolveSuper$.rebindSuper(ResolveSuper.scala:100)
	at dotty.tools.dotc.transform.ResolveSuper.superAccessors$2$$anonfun$2(ResolveSuper.scala:51)
	at scala.collection.immutable.List.map(List.scala:246)
	at dotty.tools.dotc.transform.ResolveSuper.superAccessors$3(ResolveSuper.scala:52)
	at dotty.tools.dotc.transform.ResolveSuper.$anonfun$1(ResolveSuper.scala:54)
	at scala.collection.immutable.List.flatMap(List.scala:293)
	at dotty.tools.dotc.transform.ResolveSuper.transformTemplate(ResolveSuper.scala:54)
	at dotty.tools.dotc.transform.ResolveSuper.transformTemplate(ResolveSuper.scala:42)
	at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:971)
	at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:972)
	at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:340)
	at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
	at dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:251)
	at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:394)
	at dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:404)
	at dotty.tools.dotc.transform.MegaPhase.$anonfun$1(MegaPhase.scala:409)
	at scala.collection.immutable.List.mapConserve(List.scala:472)
	at dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:409)
	at dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:356)
	at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:359)
	at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:396)
	at dotty.tools.dotc.transform.MegaPhase.transformUnit(MegaPhase.scala:415)
	at dotty.tools.dotc.transform.MegaPhase.run(MegaPhase.scala:427)
	at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:318)
	at scala.collection.immutable.List.map(List.scala:246)
	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:319)
	at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:180)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at dotty.tools.dotc.Run.runPhases$5(Run.scala:190)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:198)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:64)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:205)
	at dotty.tools.dotc.Run.compileSources(Run.scala:142)
	at dotty.tools.dotc.Run.compile(Run.scala:124)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:38)
	at dotty.tools.dotc.Driver.process(Driver.scala:194)
	at dotty.tools.dotc.Driver.process(Driver.scala:163)
	at dotty.tools.dotc.Driver.process(Driver.scala:175)
	at dotty.tools.dotc.Driver.main(Driver.scala:202)
	at dotty.tools.dotc.Main.main(Main.scala)
```
</details>

--- a/compiler/src/dotty/tools/dotc/core/Flags.scala
+++ b/compiler/src/dotty/tools/dotc/core/Flags.scala
@@ -530,6 +530,7 @@ object Flags {
 
 // ------- Other flag sets -------------------------------------
 
+  val NotConcrete: FlagSet                   = AbsOverride | Deferred
   val AbstractFinal: FlagSet                 = Abstract | Final
   val AbstractOverride: FlagSet              = Abstract | Override
   val AbstractSealed: FlagSet                = Abstract | Sealed

--- a/compiler/src/dotty/tools/dotc/typer/RefChecks.scala
+++ b/compiler/src/dotty/tools/dotc/typer/RefChecks.scala
@@ -524,10 +524,10 @@ object RefChecks {
 
       def isImplemented(mbr: Symbol) =
         val mbrType = clazz.thisType.memberInfo(mbr)
-        extension (sym: Symbol) def isConcrete = sym.exists && !sym.is(Deferred)
+        def isConcrete(sym: Symbol) = sym.exists && !sym.isOneOf(NotConcrete)
         clazz.nonPrivateMembersNamed(mbr.name)
           .filterWithPredicate(
-            impl => impl.symbol.isConcrete && mbrType.matchesLoosely(impl.info))
+            impl => isConcrete(impl.symbol) && mbrType.matchesLoosely(impl.info))
           .exists
 
       /** The term symbols in this class and its baseclasses that are
@@ -641,7 +641,11 @@ object RefChecks {
 
                     undefined(s"\n(Note that ${pa.show} does not match ${pc.show}$addendum)")
                   case xs =>
-                    undefined(s"\n(The class implements a member with a different type: ${concrete.showDcl})")
+                    undefined(
+                      if concrete.symbol.is(AbsOverride) then
+                        s"\n(The class implements ${concrete.showDcl} but that definition still needs an implementation)"
+                      else
+                        s"\n(The class implements a member with a different type: ${concrete.showDcl})")
                 }
               case Nil =>
                 undefined("")

--- a/tests/neg/i9329.check
+++ b/tests/neg/i9329.check
@@ -1,0 +1,5 @@
+-- Error: tests/neg/i9329.scala:8:6 ------------------------------------------------------------------------------------
+8 |class GrandSon extends Son // error
+  |      ^
+  |    class GrandSon needs to be abstract, since def name: => String is not defined 
+  |    (The class implements abstract override def name: => String but that definition still needs an implementation)

--- a/tests/neg/i9329.scala
+++ b/tests/neg/i9329.scala
@@ -1,0 +1,8 @@
+trait Parent {
+  def name: String
+}
+trait Son extends Parent {
+  abstract override def name = ""
+  def parentName = super.name
+}
+class GrandSon extends Son // error

